### PR TITLE
feat: add node schemas to swagger definitions for mdw

### PR DESCRIPTION
**Pull request by yaboiishere**
ref: #1850

### CODE REVIEW

**Review by davidyuk — APPROVED** [review]
It looks good, the only concern is that there would be some node schemas not used in middleware

**Comment by yaboiishere** [comment]
Yeah, I started by trying to get only specific ones, but I think It's worth it to be a little less specific, so that if we need a definition it's there. If it's a problem I can get only the specific schemas and make it easy to add more

File: scripts/swagger-docs.py [hunk 1]
```diff
@@ -23,14 +24,18 @@
...
+
 for filepath in Path(SWAGGER_DOCS_DIR).glob("*.spec.yaml"):
     with open(filepath, 'r') as stream:
         data_loaded = yaml.safe_load(stream)
         schemas = {**schemas, **data_loaded['schemas']}
         paths = {**paths, **data_loaded['paths']}
 
 swagger['paths'] = paths
-swagger['components']['schemas'] = {**swagger['components']['schemas'], **schemas}
+swagger['components']['schemas'] = {**swagger['components']['schemas'], **schemas, **node_schemas}
```
Comment (davidyuk) [inline scripts/swagger-docs.py:38]: `node_schemas` have own schemas for `Channel`, `KeyBlock`, `OracleQuery`, `Status`. Looks like it overrides mdw's schemas.

Comment (davidyuk) [reply scripts/swagger-docs.py:38]: I propose to add a code removing schemas not referenced in other places, like this
```python
schemas = swagger['components']['schemas']
schemas_count = 0
while schemas_count != len(schemas):
    schemas_count = len(schemas)
    swagger_string = str(swagger)
    for name in list(schemas):
        if '#/components/schemas/' + name not in swagger_string:
            del schemas[name]
```
It needs to be run multiple times to remove a chain of schemas referencing each other, but not referenced in other places. It reduces the schemas count from 142 to 97. This way you don't need to import specific schemas, but there won't be extra schemas in the final document.

**Review by davidyuk — COMMENTED** [review]
In general, I don't have problems with extra schemas because autorest removes them (at least I think so). But won't they be rendered in Swagger UI confusing developers?

File: docs/swagger_v3/transactions.spec.yaml [hunk 2]
```diff
@@ -59,7 +56,6 @@ schemas:
         sender_id: ak_26dopN3U2zgfJG4Ao4J4ZvLTf5mqr7WAgLAq6WxjxuSapZhQg5
         type: SpendTx
         version: 1
-        ttl: 10000
```
Comment (davidyuk) [inline docs/swagger_v3/transactions.spec.yaml:60]: This `ttl` is fine because it is in `tx` field.

**Review by davidyuk — APPROVED** [review]

**Comment by vatanasov** [comment]
Maybe look into adding a command to `scripts/do.sh`, that would copy the `node_oas3.yaml` file from the docker container to the swagger directory and then run the `swagger-docs.py`. Otherwise while developing you need to get the node file from somewhere (which is kinda confusing) and than it will probably be left stale, unless you remember to update it manually the next time you are working on the openapi schemas. It's not that big of a problem, but having it automated would be less error-prone and provide better development experience.

**Comment by davidyuk** [comment]
I found out that we can pass an external url to `$ref` 🙃 
https://swagger.io/docs/specification/using-ref/

**Comment by yaboiishere** [comment]
That was my first aproach, but then we would have to figure out what the version of the node that's running with the mdw, so that we reference the correct swagger.

**Review by sborrazas — APPROVED** [review]

**Pull request merged**